UniWatch/Controllers/StudentController.cs

```diff
@@ -0,0 +1,191 @@
+using UniWatch.Models;
+using UniWatch.ViewModels;
+
+namespace UniWatch.Controllers
+{
+    public class StudentController : Controller
+    {
+        private IDataAccess _dataAccess;
+
+        public StudentController() : this(new DataAccess.DataAccess())
+        { }
+
+        public StudentController(IDataAccess dataAccess)
+        {
+            _dataAccess = dataAccess;
+        }
+
+        // GET: /Student/Index?classId=1
+        /// <summary>
+        /// Display a list of students in a class
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        [HttpGet]
+        public ActionResult Index(int classId)
+        {
+            var @class = _dataAccess.ClassManager.GetById(classId);
+            if (@class == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            var students = _dataAccess.ClassManager.GetEnrolledStudents(classId);
+            return View(new ViewModels.EnrollIndexViewModel
+            {
+                Class = @class,
+                EnrolledStudents = students
+            });
+        }
+
+        /// <summary>
+        /// Returns the view to enroll a student in a class. 
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        [HttpGet]
+        public ActionResult Enroll(int classId)
+        {
+            var @class = _dataAccess.ClassManager.GetById(classId);
+            if (@class == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            EnrollStudentViewModel viewModel = new EnrollStudentViewModel
+            {
+                Class = @class,
+                StudentsFound = null
+            };
+
+            return View(viewModel);
+        }
+
+        /// <summary>
+        /// Returns the view to enroll a student in a class. Searches for students
+        /// that match the given id
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        /// <param name="studentId">The student search parameter</param>
+        [HttpPost]
+        public ActionResult Search(int classId, string studentId)
+        {
+            var @class = _dataAccess.ClassManager.GetById(classId);
+            if (@class == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            var found = _dataAccess.UserManager.SearchStudent(studentId);
+            var enrolled = _dataAccess.ClassManager.GetEnrolledStudents(classId);
+
+            // Set of student ids (for enrolled students)
+            HashSet<int> enrolledSet = new HashSet<int>();
+            foreach (var student in enrolled)
+                enrolledSet.Add(student.Id);
+
+            EnrollStudentViewModel viewModel = new EnrollStudentViewModel
+            {
+                Class = @class
+            };
```

Comment:
Is is not needed, but if you know the size your List is going to be you can initialize the size here.

```
EnrollStudentViewModel viewModel = new EnrollStudentViewModel
{
  Class = @class,
  StudentsFound = new List<StudentFound>(found.Count)
};
```